packages/react-data-grid-examples/src/scripts/example30-efficient-windowing.js

```diff
@@ -0,0 +1,148 @@
+import ReactDataGrid from 'react-data-grid';
+import exampleWrapper from '../components/exampleWrapper';
+import React from 'react';
+
+const COL_WIDTH = 200;
+const ROW_HEIGHT = 35;
+const HEADER_HEIGHT = 35;
+
+const createColumns = (numberCols) => [...Array(numberCols).keys()].map(i => {
+  if (i < 3) {
+    return {
+      key: `col${i}`,
+      name: `col${i}`,
+      width: COL_WIDTH,
+      locked: true
+    }
+  }
+  return {
+    key: `col${i}`,
+    name: `col${i}`,
+    width: COL_WIDTH
+  }
+});
+
+const createRows = (numberRows) => [...Array(numberRows).keys()].map(i => {
+  return [...Array(100).keys()].reduce((row, j) => ({...row, ...{[`col${j}`]: `row ${i} col ${j}`} }), {});
+});
+
+const RenderWindow = ({ isScrolling, colOverscanEndIdx, colOverscanStartIdx, rowOverscanEndIdx, rowOverscanStartIdx, rowVisibleStartIdx, rowVisibleEndIdx, colVisibleStartIdx, colVisibleEndIdx  }) => {
+  const topOffset = (rowOverscanStartIdx - rowVisibleStartIdx) * ROW_HEIGHT;
+  const marginTop = HEADER_HEIGHT + topOffset;
+  const marginLeft = (colOverscanStartIdx - colVisibleStartIdx) * COL_WIDTH;
+  const height = (rowOverscanEndIdx - rowOverscanStartIdx) * ROW_HEIGHT;
+  const width = (colOverscanEndIdx - colOverscanStartIdx) * COL_WIDTH;
+  return (isScrolling ? <div 
+    style={{ position: 'absolute', backgroundColor: 'orange', opacity: 0.7, height, width, marginTop, marginLeft, zIndex: 2, pointerEvents: 'none' }} >
+    </div> : null);
+};
+
+const NumberInput = ({name, value, onChange}) => {
+  return (
+  <div className="col-md-2">
+  <div className="input-group">
+  <span className="input-group-addon" id="basic-addon1">{name}</span>
+  <input type="text" className="form-control" defaultValue={value} onChange={onChange}  />
```

Comment:
Am actually removing this whole example as it was more for my own debugging purposes around columns/row rendering rather than to demonstrate an actual grid feature.